group/assignment6/RearrangingCars.java

```diff
@@ -0,0 +1,75 @@
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class RearrangingCars {
+
+    public static ArrayList<Car> reArrangeCars(ArrayList<Car> cars){
+        if(cars.size() == 0)
+            return null;
+
+        boolean[] N = markInitialPlaces(cars);
```

Comment:
Just a small comment on the naming. "N" doesn't say anything about the meaning of this variable. Additionally I think in Java it is common to start variable names with lower case.